phantazm-api/src/main/java/com/github/phantazmnetwork/api/config/VariantConfigProcessor.java

```diff
@@ -0,0 +1,61 @@
+package com.github.phantazmnetwork.api.config;
+
+import com.github.phantazmnetwork.commons.AdventureConfigProcessors;
+import com.github.steanky.ethylene.core.ConfigElement;
+import com.github.steanky.ethylene.core.collection.ConfigNode;
+import com.github.steanky.ethylene.core.collection.LinkedConfigNode;
+import com.github.steanky.ethylene.core.processor.ConfigProcessException;
+import com.github.steanky.ethylene.core.processor.ConfigProcessor;
+import net.kyori.adventure.key.Key;
+import net.kyori.adventure.key.Keyed;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A {@link ConfigProcessor} which processes {@link TValue}s.
+ * This can be used to process inheritance and identifies variants based on {@link Keyed#key()}.
+ * @param <TValue> The type of {@link TValue} to process
+ */
+public class VariantConfigProcessor<TValue extends Keyed> implements ConfigProcessor<TValue> {
+
+    private static final ConfigProcessor<Key> KEY_PROCESSOR = AdventureConfigProcessors.key();
+
+    private final Map<Key, ConfigProcessor<? extends TValue>> subProcessors;
+
+    /**
+     * Creates a new {@link VariantConfigProcessor}.
+     * @param subProcessors A map of {@link Key}s to {@link ConfigProcessor}s to delegate processing to
+     */
+    public VariantConfigProcessor(@NotNull Map<Key, ConfigProcessor<? extends TValue>> subProcessors) {
```

Comment:
might be better to use Function<Key, ConfigProcessor<? extends TValue>> instead of a map, gives caller more flexibility

Reply:
sure but I can't imagine a case where that would be used over a map?